### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by nashjain**
# Description

In SSEHeaderFilter, removed unwanted IOException
In Stat.kt and StatsCollectionOrchestrator.kt - Specified type explicitly to avoid unchecked nullability issues from a declaration where the type was inferred from a platform call.
In StatEventHandler, replaced .map(){}.toMap() with associateWith()
In StatTimeTrigger, made timer private to avoid indecent exposure code smell. Also, public access specifier was redundant for shutdownTimer()

Fixes # (issue)

## Type of change

- [x] Refactoring

# How Has This Been Tested?

- [x] Ran all the unit tests

### CODE REVIEW

File: backend/edge-full/src/main/kotlin/io/featurehub/edge/stats/StatsCollectionOrchestrator.kt [hunk 1]
```diff
@@ -24,9 +24,9 @@ class StatsCollectionOrchestrator @Inject constructor(private val publisher: Sta
   var maxApiKeysPerPublish: Long? = 300
 
   companion object Prometheus {
-    val publishTimeHistogram = Histogram.build("edge_publish_time", "Time taken to publish stats to NATS").register()
-    val successfulPublishing = Counter.build("edge_publish_success", "Number of successes for publishing").register()
-    val failedPublishing = Counter.build("edge_publish_failure", "Number of failures for publishing").register()
+    val publishTimeHistogram: Histogram = Histogram.build("edge_publish_time", "Time taken to publish stats to NATS").register()
```
Comment (rvowles) [inline backend/edge-full/src/main/kotlin/io/featurehub/edge/stats/StatsCollectionOrchestrator.kt:27]: why declare the types here? they are implicit from the return value?

Comment (nashjain) [reply backend/edge-full/src/main/kotlin/io/featurehub/edge/stats/StatsCollectionOrchestrator.kt:27]: The return type is not implicit. See the warning in IntelliJ IDE

IDE_Warning

Comment (rvowles) [reply backend/edge-full/src/main/kotlin/io/featurehub/edge/stats/StatsCollectionOrchestrator.kt:27]: Right, that makes sense now.

**Review by rvowles — APPROVED** [review]

**Pull request merged**